Support sending and logging in with an unblock code

kewisch opened the pull request:
This should fix it, I've tested it and it works in practice.

The unit test I have not tested and I would appreciate some guidance on. I don't think it is possible to put the server into a state where the account is blocked so the flow can be tested.

kewisch commented:
Ok now, the remaining test failures are probably not mine. Let me know if there is anything I should change or do to get this merged.

rfk commented:
Thanks @kewisch!  Just to le tyou know, I'm on holiday for the coming week so won't be able to take a proper look until then, but I'll keep this in my inbox to review when I return.

rfk commented on fxa/tests/test_core.py:
```diff
@@ -176,6 +176,32 @@ def wait_for_email(m):
+        self.assertTrue(len(code) > 0)
+
+        # XXX TODO: the live server doesn't block the account, so we can't
+        # really test login flow with an unblock code
```
Actually, since this talks to the dev server by default, I believe it will block login attempts for addresses that match `/^(block.*@restmail\.net)$/`.  If you create the TestEmailAccount to match that pattern then the I think we could get a working test here.

kewisch commented:
The commented out test works for me locally when running it directly. I wasn't able to complete a full test run due to timeouts and request errors, I hope travis does a better job. I've also fixed a few flake8 errors and rebased against master.

kewisch commented:
The failing tests seem to be something else. I got further locally by replacing restmail.com with restmail.net in code I didn't touch, but another test is timing out trying to contact `verifier.login.persona.org`. @rfk can you check and see if this is something else?

rfk commented:
> but another test is timing out trying to contact verifier.login.persona.org

That's definitely unrelated, and due to browserid infra being shut down.  I'm going to make a new release of PyBrowserID that will hopefully fix the problem by defaulting to a local verifier rather than the (no longer running) remote verifier.

rfk commented:
Kicking off a rebuild, let's see if the new PyBrowserID release helps things...

Natim commented:
> Kicking off a rebuild, let's see if the new PyBrowserID release helps things...

Yes much better only a mocked tests failed and I guess we can update it since it seems related to that PR.

rfk commented:
LGTM, thanks @kewisch!
